### ISSUE TITLE
Unable to deploy : Dropstack asks to login when I'm already logged in.

### ISSUE THREAD

**Issue by hemangsk** [CodeCommission/dropstack-cli#40]
Details: When I try to do dropstack deploy, it shows up a message ```Credentials not found. Use dropstack login <url> to verifiy your credentials.``` even If I'm already logged in. That's weird and has started happening since 3 days. It'll be really awesome if you can help me figure out what's going wrong. 

StackTrace:

```
Hemang ~/PycharmProjects/cs $ dropstack ls
Deployments for hemangsk@gmail.com on https://api.cloud.dropstack.run

droTYPE    | NUM # | ID/NAME   | URL                                          
Nothing deployed
Hemang ~/PycharmProjects/cs $ dropstack deploy
Deploying - for hemangsk@gmail.com on https://api.cloud.dropstack.run
⣇ Deploying [3.88s]
Credentials not found. Use dropstack login <url> to verifiy your credentials.
Hemang ~/PycharmProjects/cs $ dropstack info
Account information for hemangsk@gmail.com on https://api.cloud.dropstack.run

   ┌──────────────────────────────────────────────────────┐
   │                                                      │
   │   Account ID     : hemangsk@gmail.com                │
   │   Server Version : 2.5.7                             │
   │   Client Version : 2.5.7                             │
   │   Server URL     : https://api.cloud.dropstack.run   │
   │   Plan (beta)    : free                              │
   │                                                      │
   └──────────────────────────────────────────────────────┘

Hemang ~/PycharmProjects/cs $ dropstack deploy
Deploying - for hemangsk@gmail.com on https://api.cloud.dropstack.run
⣇ Deploying [3.76s]
Credentials not found. Use dropstack login <url> to verifiy your credentials.
Hemang ~/PycharmProjects/cs $ 
```
I'm using 
```
dropstack 2.5.7
npm 5.0.3
macOS Sierra
```
Thanks!

**Comment by MikeBild** [comment]
Did you tried to use dropstack logout and relogin with dropstack login?

**Comment by hemangsk** [comment]
Yeah I tried doing logout and logging in yesterday but that didn't fix this issue. 
But today it's showing an error when I try to login after logging out
```
Hemang ~/PycharmProjects/cs $ dropstack login
Account information for - on https://api.cloud.dropstack.run
Enter your email and password to log in or sign up for free!

email: hemangsk@gmail.com
password: ********


An unexpected error occurred!
Error: request to https://api.cloud.dropstack.run/auth/signup failed, reason: getaddrinfo ENOTFOUND api.cloud.dropstack.run api.cloud.dropstack.run:443
```
This was not coming yesterday.

**Comment by MikeBild** [comment]
Could you try it again please!

**Comment by hemangsk** [comment]
Yes the auth/signup issue is resolved now :) 

While deploying, the credentials error comes up as in the following output of ```dropstack deploy -o``` :
```
Hemang ~/PycharmProjects/cs $ dropstack logout
Logout for hemangsk@gmail.com on https://api.cloud.dropstack.run

   ┌─────────────────┐
   │                 │
   │   Logged out!   │
   │                 │
   └─────────────────┘

Hemang ~/PycharmProjects/cs $ dropstack login
Account information for - on https://api.cloud.dropstack.run
Enter your email and password to log in or sign up for free!

email: hemangsk@gmail.com
password: ********


   ┌──────────────────────────────────────────────────────┐
   │                                                      │
   │   User hemangsk@gmail.com logged in successful!      │
   │                                                      │
   │   Account ID     : hemangsk@gmail.com                │
   │   Server Version : 2.5.9                             │
   │   Client Version : 2.5.7                             │
   │   Server URL     : https://api.cloud.dropstack.run   │
   │   Plan (beta)    : free                              │
   │                                                      │
   └──────────────────────────────────────────────────────┘

Hemang ~/PycharmProjects/cs $ dropstack deploy -o 
Deploying - for hemangsk@gmail.com on https://api.cloud.dropstack.run
-------- packing ...
-------- uploading [0.02 MB]
-------- 100.00% uploaded [0.02 MB]
⢺ Processing [1.42s]jpcjofnf deploying 

Credentials not found. Use dropstack login <url> to verifiy your credentials.
```

**Comment by MikeBild** [comment]
Thank you! There is throwing an unpacking (zip) error of your current build package. Is it possible to extract or share your project structure to reproduce this error on my site?

**Comment by hemangsk** [comment]
Hey Mike, thanks a lot for your help. My project is a basic django project with following structure
```
cs
    - urls.py
    - views.py

text
    - urls.py
    - models.py
    - utilities.py
    - views.py
db.sqlite3
manage.py
requirements.txt
Dockerfile
``` 

Yesterday I tried a lot to make a minimal reproduction of this error. What I found is, 
- I was able to successfully deploy a fresh Django project but not the one old one(which is the actual project). Then I created a fresh Django project, moved my code into that project and it got deployed successfully
- Then I renamed the app of my old project, regenerated the migrations and db.sqlite3 file. It got deployed successfully :)

I am yet to confirm the reason why this might have happened, and will post here asap.

**Comment by MikeBild** [comment]
Thank you for your investigations. One more question: Is it possible there are some hidden files in the old project?

**Comment by hemangsk** [comment]
Yeah I just checked, the following hidden files are present in the root directory apart from .git and .gitignore there is a .idea generated by intellij.
I suspect that the db.sqlite3 file might be creating a conflict with the one already present on server, but I'm not completely sure on this.

Latest Deployment Trace 
```
Hemang ~/PycharmProjects/cs $ dropstack deploy -o
Deploying sqpcrgar for hemangsk@gmail.com on https://api.cloud.dropstack.run
sqpcrgar packing ...
sqpcrgar uploading [0.00 MB]
sqpcrgar 163840.00% uploaded [0.02 MB]
⢺ Processing [1.54s]sqpcrgar 247820.00% uploaded [0.02 MB]
sqpcrgar deploying 
sqpcrgar analysing 
sqpcrgar analysed 
sqpcrgar building 
sqpcrgar building Step 1/9 : FROM frolvlad/alpine-python3
sqpcrgar building  ---> af6c5a49da72
sqpcrgar building Step 2/9 : RUN apk add --no-cache sqlite
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> 3ebc7ad65dbf
sqpcrgar building Step 3/9 : WORKDIR /src
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> 9060e16eeb3e
sqpcrgar building Step 4/9 : ADD . .
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> f910e6c4e62e
sqpcrgar building Step 5/9 : EXPOSE 80
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> da3d33022078
sqpcrgar building Step 6/9 : RUN pip3 install -r requirements.txt
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> 6ae1cfec1beb
sqpcrgar building Step 7/9 : RUN python3 manage.py makemigrations
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> cf9abaab471f
sqpcrgar building Step 8/9 : RUN python3 manage.py migrate
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> ceb8bd690660
sqpcrgar building Step 9/9 : ENTRYPOINT python3 manage.py runserver 0.0.0.0:80
sqpcrgar building  ---> Using cache
sqpcrgar building  ---> 083eb2559fb4
sqpcrgar building 
sqpcrgar building Successfully built 083eb2559fb4
sqpcrgar building Successfully tagged api.cloud.dropstack.run:5000/sqpcrgar:latest
sqpcrgar builded 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar publishing 
sqpcrgar published 
sqpcrgar starting 
sqpcrgar started 
sqpcrgar registration 
Deployment registrated [5.31s]

   ┌────────────────────────────────────────────────────────────┐
   │                                                            │
   │   Deployment sqpcrgar completed - 1 running instance(s)!   │
   │                                                            │
   │   • URL: https://sqpcrgar.cloud.dropstack.run              │
   │                                                            │
   │   URL copied to clipboard.                                 │
   │                                                            │
   └────────────────────────────────────────────────────────────┘
```

**Comment by MikeBild** [comment]
Ok, thank you! Normally there shouldn't conflict with something "old". Every deploy creates a new immutable image for the deploy. 

BTW: It's possible to exclude files and folders e.g. `.idea`, etc.

e.g. - `dropstack deploy -x **/.idea/* -x **/.git/*`

Did you try this to tackle the upload/unzip problem?

**Comment by MikeBild** [comment]
Any news?

**Comment by hemangsk** [comment]
Hey Mike, sorry for the delay in response. I tried to reproduce this upload/unzip issue, but i am unable to reproduce it. I don't find any issue with the deployment right now :) It was really weird why it was happening earlier because with the same set of hidden files, I managed to make a successful deployment for a fresh project (with the dropstack deploy -x option as well and without it as well) , and after renaming a module in my old django project its deployment was also successful.  And its working perfectly with new and old projects, tried it rigorously.

Thanks a lot for your support 😊  dropstack is awesome
